include/morphio/mut/soma.h

```diff
@@ -75,7 +73,15 @@ const std::vector<float>& Soma::diameters() const noexcept {
 }
 
 inline SomaType Soma::type() const noexcept {
-    return _somaType;
+    // this function seems not complete, see /src/readers/morphologySWC.cpp, line 217 for a more complete one
```

Comment:
I would rather use a `switch` instead of allocating a `std::map` all the time and rely on an expensive exception raised by `std::map::at`.

Something like:
```c++
switch(points().size()) {
  case 0:
  case 2:
    return SOMA_UNDEFINED;
  case 1:
    return SOMA_SINGLE_POINT;
  default:
    return SOMA_SIMPLE_CONTOUR;
}
```

Besides the member function can then be marked with `noexcept` specifier.

Reply:
what do you think about creating a static std::array<SomaType> v = {SOMA_UNDEFINED, SOMA_SINGLE_POINT, SOMA_UNDEFINED}; and pick from that if nPoints are inside the range? There is only one size check; it should be faster